pymongo/server_api.py

```diff
@@ -19,28 +19,62 @@
 MongoDB Versioned API
 =====================
 
+Starting in MongoDB 5.0, applications can specify the server API version
+to use when creating a :class:`~pymongo.mongo_client.MongoClient`. Doing so
+ensures that the driver behaves in a manner compatible with that server API
+version, regardless of the server's actual release version.
+
+Declaring an API Version
+````````````````````````
+
 To configure MongoDB Versioned API, pass the ``server_api`` keyword option to
 :class:`~pymongo.mongo_client.MongoClient`::
 
-    from pymongo.mongo_client import MongoClient
-    from pymongo.server_api import ServerApi
-
-    client = MongoClient(server_api=ServerApi('1'))
+    >>> from pymongo.mongo_client import MongoClient
+    >>> from pymongo.server_api import ServerApi
+    >>>
+    >>> # Declare API version "1" for MongoClient "client"
+    >>> server_api = ServerApi('1')
+    >>> client = MongoClient(server_api=server_api)
+
+The declared API version is applied to all commands run through ``client``,
+including those sent through the generic command helper. Specifying versioned
+API options in the command document **and** declaring an API version
+on the client is not supported and will lead to undefined behaviour.
+To run any command without declaring a server API version or using a different
+API version, create a separate :class:`~pymongo.mongo_client.MongoClient`
+instance.
 
 Note that Versioned API requires MongoDB >=5.0.
 
 Strict Mode
 ```````````
 
-When ``strict`` mode is configured, commands that are not supported in the
-given :attr:`ServerApi.version` will fail. For example::
+Configuring ``strict`` mode will cause the MongoDB server to reject all
+commands that are not part of the declared :attr:`ServerApi.version`. This
+includes command options and aggregation pipeline stages.
+
+For example::
 
-    >>> client = MongoClient(server_api=ServerApi('1', strict=True))
+    >>> server_api=ServerApi('1', strict=True)
+    >>> client = MongoClient(server_api=server_api)
     >>> client.test.command('count', 'test')
     Traceback (most recent call last):
     ...
     pymongo.errors.OperationFailure: Provided apiStrict:true, but the command count is not in API Version 1, full error: {'ok': 0.0, 'errmsg': 'Provided apiStrict:true, but the command count is not in API Version 1', 'code': 323, 'codeName': 'APIStrictError'
 
+Detecting API Deprecations
+``````````````````````````
+
+The ``deprecationErrors`` option can be used to enable command failures
+when using functionality that is deprecated from the configured
+:attr:`ServerApi.version`. For example::
+
+    >>> server_api=ServerApi('1', deprecation_errors=True)
+    >>> client = MongoClient(server_api=server_api)
+
+Note that at the time of this writing, no deprecated APIs exist.
```

Comment:
Do we have an example of what a deprecation would look like? Would it be an OperationFailure error?

Reply:
I was not able to find one. The closest I got was finding this information about the error code and category that will be returned when a command is Deprecated is [here](https://github.com/mongodb/mongo/commit/70fc2eb27835c2fd0eb3150590418692202e3d75).

```
error_categories:
    ...
    - VersionedAPIError

error_codes: 
    ...
    - {code: 322, name: APIVersionError, categories: [VersionedAPIError]}
    - {code: 323, name: APIStrictError, categories: [VersionedAPIError]}
    - {code: 324, name: APIDeprecationError, categories: [VersionedAPIError]}
```

So I _think_ it'd be surfaced to the user as an `OperationFailure`?

Reply:
Yes
```
pymongo.errors.OperationFailure: API version must be "1", full error: {'ok': 0.0, 'errmsg': 'API version must be "1"', 'code': 322, 'codeName': 'APIVersionError', '$clusterTime': {'clusterTime': Timestamp(1617170093, 1), 'signature': {'hash': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 'keyId': 0}}, 'operationTime': Timestamp(1617170093, 1)}
```

Reply:
```
>>> client.db.command(command={'testDeprecationInVersion2': 1, 'apiVersion': '2', 'apiDeprecationErrors': True})
...
pymongo.errors.OperationFailure: Provided apiDeprecationErrors:true, but the command testDeprecationInVersion2 is deprecated in API Version 2, full error: {'ok': 0.0, 'errmsg': 'Provided apiDeprecationErrors:true, but the command testDeprecationInVersion2 is deprecated in API Version 2', 'code': 324, 'codeName': 'APIDeprecationError', '$clusterTime': {'clusterTime': Timestamp(1617170403, 1), 'signature': {'hash': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 'keyId': 0}}, 'operationTime': Timestamp(1617170403, 1)}
```

Reply:
I'm reluctant to include this as an example because the server needs to be started with special testing parameters enabled to be able to see this behavior.